static/src/js/util/accessMethods/buildAccessMethods/buildEsiEcho.js

```diff
@@ -0,0 +1,65 @@
+import { camelCase } from 'lodash-es'
+import { generateFormDigest } from '../generateFormDigest'
+
+/**
+ * Builds the ESI or Echo Ordering access method
+ * @param {object} serviceItem serviceItem in the Collection Metadata
+ * @param {boolean} disabledOrdering true if ordering is disabled
+ * @returns {object} Access method for ESI or Echo Orders
+ */
+export const buildEsiEcho = (serviceItem, disableOrdering) => {
+  const accessMethods = {}
+  // Only process orderOptions if the service type uses orderOptions
+  // Do not include access if orders are disabled
+
+  const {
+    orderOptions,
+    type: serviceType,
+    url,
+    maxItemsPerOrder
+  } = serviceItem
+
+  if (disableOrdering !== 'true') {
+    const { urlValue } = url
+
+    const { items: orderOptionsItems } = orderOptions
+    if (orderOptionsItems === null) return {}
+
+    let esiIndex = 0
```

Comment:
This does not fix the index issue, these indexes will still overwrite existing esi or echo methods coming from multiple UMM-S records